pkg/solana/chainreader/chain_reader.go

```diff
@@ -114,123 +111,73 @@ func (s *SolanaChainReaderService) GetLatestValue(ctx context.Context, readIdent
 		return fmt.Errorf("%w: no contract for read identifier %s", types.ErrInvalidType, readIdentifier)
 	}
 
-	addressMappings, err := decodeAddressMappings(vals.address)
-	if err != nil {
-		return fmt.Errorf("%w: %s", types.ErrInvalidConfig, err)
-	}
-
-	addresses, ok := addressMappings[vals.readName]
-	if !ok {
-		return fmt.Errorf("%w: no addresses for readName %s", types.ErrInvalidConfig, vals.readName)
+	batch := []call{
+		{
+			ContractName: vals.contract,
+			ReadName:     vals.readName,
+			Params:       params,
+			ReturnVal:    returnVal,
+		},
 	}
 
-	bindings, err := s.bindings.GetReadBindings(vals.contract, vals.readName)
+	results, err := doMethodBatchCall(ctx, s.client, s.bindings, batch)
 	if err != nil {
 		return err
 	}
 
-	if len(addresses) != len(bindings) {
-		return fmt.Errorf("%w: addresses and bindings lengths do not match", types.ErrInvalidConfig)
+	if len(results) != len(batch) {
+		return fmt.Errorf("%w: unexpected number of results", types.ErrInternal)
 	}
 
-	// if the returnVal is not a *values.Value, run normally without using the ptrToValue
-	ptrToValue, isValue := returnVal.(*values.Value)
-	if !isValue {
-		return s.runAllBindings(ctx, bindings, addresses, params, returnVal)
-	}
-
-	// if the returnVal is a *values.Value, create the type from the contract, run normally, and wrap the value
-	contractType, err := s.bindings.CreateType(vals.contract, vals.readName, false)
-	if err != nil {
-		return err
-	}
-
-	if err = s.runAllBindings(ctx, bindings, addresses, params, contractType); err != nil {
-		return err
-	}
-
-	value, err := values.Wrap(contractType)
-	if err != nil {
-		return err
+	if results[0].err != nil {
+		return fmt.Errorf("%w: %s", types.ErrInternal, results[0].err)
 	}
 
-	*ptrToValue = value
-
 	return nil
```

Comment:
Don't we still need to set `returnVal.(*values.Value)`?

Reply:
The type casting creates the pointer to the value. Setting the value on line 157 is correct. You can see the implementation in evm is the same.